lib/gpu_memory_service/tests/test_gms_write_publication.py

```
@@ -76,66 +162,954 @@ def prune(_allocator, *, referenced_allocation_ids):
+            [ModuleTensorBinding("second", ModuleTensorKind.ATTRIBUTE)],
+        ),
+    ]
+    replacements = {
+        id(first): torch.full_like(first, 20),
+        id(second): torch.full_like(second, 30),
+    }
+    first_storage = first.untyped_storage()._cdata
+    second_storage = second.untyped_storage()._cdata
+    second_ref = weakref.ref(second)
+
+    with pytest.raises(RuntimeError, match="second.*weakref"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)